src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -54,18 +58,36 @@ void KeyboardManagerState::ClearOSLevelShortcuts()
     osLevelShortcutReMap.clear();
 }
 
+// Function to clear the Keys remapping table.
+void KeyboardManagerState::ClearSingleKeyRemaps()
+{
+    singleKeyReMap.clear();
+}
+
 // Function to add a new OS level shortcut remapping
 void KeyboardManagerState::AddOSLevelShortcut(const std::vector<DWORD>& originalSC, const std::vector<WORD>& newSC)
 {
     osLevelShortcutReMap[originalSC] = std::make_pair(newSC, false);
 }
 
+// Function to add a new OS level shortcut remapping
+void KeyboardManagerState::AddSingleKeyRemap(const DWORD& originalKey, const WORD& newRemapKey)
+{
+    singleKeyReMap[originalKey] = newRemapKey;
+}
+
 // Function to set the textblock of the detect shortcut UI so that it can be accessed by the hook
 void KeyboardManagerState::ConfigureDetectShortcutUI(const TextBlock& textBlock)
 {
     currentShortcutTextBlock = textBlock;
 }
 
+// Function to set the textblock of the detect remap key UI so that it can be accessed by the hook
+void KeyboardManagerState::ConfigureDetectSingleKeyRemapUI(const winrt::Windows::UI::Xaml::Controls::TextBlock& textBlock)

```

Comment:
[nit] We should keep the way we reference `TextBlock` consistent in the file. Either we should fully qualify it as `winrt::Windows::UI::Xaml::Controls::TextBlock` everywhere in the file or just `TextBlock` everywhere in the file.

Reply:
Removed the fully qualified namespace for `TextBlock`